cppcache/src/TcrConnection.cpp

```diff
@@ -691,51 +691,70 @@ char* TcrConnection::sendRequest(const char* buffer, size_t len,
   return readMessage(recvLen, receiveTimeoutSec, true, &opErr, false, request);
 }
 
-void TcrConnection::sendRequestForChunkedResponse(
-    const TcrMessage& request, size_t len, TcrMessageReply& reply,
-    std::chrono::microseconds sendTimeoutSec,
-    std::chrono::microseconds receiveTimeoutSec) {
-  auto msgType = request.getMessageType();
-  switch (msgType) {
-    case TcrMessage::QUERY:
-    case TcrMessage::QUERY_WITH_PARAMETERS:
-    case TcrMessage::EXECUTECQ_WITH_IR_MSG_TYPE:
-    case TcrMessage::GETDURABLECQS_MSG_TYPE:
-    case TcrMessage::EXECUTE_FUNCTION:
-    case TcrMessage::EXECUTE_REGION_FUNCTION:
-    case TcrMessage::EXECUTE_REGION_FUNCTION_SINGLE_HOP: {
-      receiveTimeoutSec = reply.getTimeout();
-      sendTimeoutSec = reply.getTimeout();
-      break;
-    }
-    default:
-      break;
-  }
-
+std::chrono::microseconds TcrConnection::sendWithTimeouts(
+    const char* data, size_t len, std::chrono::microseconds sendTimeout,
+    std::chrono::microseconds receiveTimeout) {
   std::chrono::microseconds timeSpent{0};
```

Comment:
the diff is a bit dodgy to look at... the new method `sendWithTimeouts` is an extraction of the next several lines, no?

Reply:
Yeah, all the new methods are just extractions of blocks of code